lib/ansi.nix

```diff
@@ -0,0 +1,42 @@
+# This exports a basic set of ANSI escape sequences to colorize terminal output.
+
+{ ... }:
+
+let
+  # Escape character. "\e" or similar encodings currently not supported by Nix.
+  esc = "";
+  reset = "${esc}[0m";
+  style = {
+    bold = "${esc}[1m";
+    italic = "${esc}[3m";
+  };
+  colorCode = {
+    black = 0;
+    red = 1;
+    green = 2;
+    yellow = 3;
+    blue = 4;
+    magenta = 5;
+    cyan = 6;
+    white = 7;
+  };
+  color = {
+    fg = builtins.mapAttrs (name: value: "${esc}[3${toString value}m") colorCode;
+    bg = builtins.mapAttrs (name: value: "${esc}[4${toString value}m") colorCode;
+  };
```

Comment:
I think setting the background color is useless enough that we might not even want to have it, even just for completeness.

Reply:
I have no strong opinion, but please have a look at my comment below: https://github.com/NixOS/nixpkgs/pull/267189#issuecomment-1809660530

Reply:
Update: As written by @infinisil in https://github.com/NixOS/nixpkgs/pull/267189#issuecomment-1854687046

Functionality is/should be primarily for Nix, not for shell scripts.